sql/api/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBaseParser.g4

```diff
@@ -1337,6 +1341,15 @@ colType
     : colName=errorCapturingIdentifier dataType (errorCapturingNot NULL)? commentSpec?
     ;
 
+tableElementList
+    : tableElement (COMMA tableElement)*
+    ;
+
+tableElement
+    : tableConstraintDefinition
+    | colDefinition
```

Comment:
So `tableConstraintDefinition, tableConstraintDefinition` will be also parsed as `tableElementList`? I think it is required to be `colDefinition (tableConstraintDefinition)*` for each `tableElement`?

Reply:
This syntax is from the ANSI SQL standard.
A `tableElement` can be `colDefinition` or `tableConstraintDefinition`. A  `colDefinition` can contain a column constraint after column name and column data type.

Reply:
Oh, I see. The constraint in ` colDefinition` is column-level, a `tableConstraintDefinition` is table-level.